apps/digibyteprice/digibyte_price.star

```diff
@@ -0,0 +1,348 @@
+                icon = "toggle-on",
+                default = DEFAULT_SHOW_SECOND_CURRENCY,
+            ),
+            schema.Toggle(
+                id = "sats_toggle",
+                name = "Display SATS",
+                desc = "Toggle displaying price in SATS.",
+                icon = "toggle-on",
+                default = DEFAULT_SHOW_SATS,
+            ),
+            schema.Toggle(
+                id = "country_toggle",
+                name = "Display Country",
+                desc = "Toggle displaying country code. Useful when displaying two dollar currencies together (e.g. USD and AUD).",
+                icon = "toggle-on",
+                default = DEFAULT_SHOW_COUNTRY,
+            ),
+        ],
+    )
+
+def main(config):
+    DIGIBYTE_PRICE_URL = "https://api.coingecko.com/api/v3/simple/price?ids=digibyte&vs_currencies=aud%2Ccad%2Ceur%2Cgbp%2Csats%2Cusd"
+
+    first_currency = CURRENCY_LIST.get(config.get("first_currency"), CURRENCY_LIST[DEFAULT_FIRST_CURRENCY])
+    second_currency = CURRENCY_LIST.get(config.get("second_currency"), CURRENCY_LIST[DEFAULT_SECOND_CURRENCY])
+    first_currency_toggle = config.bool("first_currency_toggle", DEFAULT_FIRST_CURRENCY)
+    second_currency_toggle = config.bool("second_currency_toggle", DEFAULT_SECOND_CURRENCY)
+    sats_toggle = config.bool("sats_toggle", DEFAULT_SHOW_SATS)
+    country_toggle = config.bool("country_toggle", DEFAULT_SHOW_COUNTRY)
+
+    # LOOKUP CURRENT PRICES (OR RETRIEVE FROM CACHE)
+
+    # Get current prices from cache
+    dgb_price_aud_cached = cache.get("dgb_price_aud")
+    dgb_price_cad_cached = cache.get("dgb_price_cad")
+    dgb_price_eur_cached = cache.get("dgb_price_eur")
+    dgb_price_gbp_cached = cache.get("dgb_price_gbp")
+    dgb_price_sats_cached = cache.get("dgb_price_sats")
+    dgb_price_usd_cached = cache.get("dgb_price_usd")
+
+    if dgb_price_usd_cached != None:
+        print("Hit! Displaying cached data.")
+        dgb_price_aud = float(dgb_price_aud_cached)
+        dgb_price_cad = float(dgb_price_cad_cached)
+        dgb_price_eur = float(dgb_price_eur_cached)
+        dgb_price_gbp = float(dgb_price_gbp_cached)
+        dgb_price_sats = float(dgb_price_sats_cached)
+        dgb_price_usd = float(dgb_price_usd_cached)
+    else:
+        print("Miss! Calling CoinGecko API.")
+        dgbquery = http.get(DIGIBYTE_PRICE_URL)
+        if dgbquery.status_code != 200:
+            fail("Coingecko request failed with status %d", dgbquery.status_code)
+
+        dgb_price_aud = dgbquery.json()["digibyte"]["aud"]
+        dgb_price_cad = dgbquery.json()["digibyte"]["cad"]
+        dgb_price_eur = dgbquery.json()["digibyte"]["eur"]
+        dgb_price_gbp = dgbquery.json()["digibyte"]["gbp"]
+        dgb_price_sats = dgbquery.json()["digibyte"]["sats"]
+        dgb_price_usd = dgbquery.json()["digibyte"]["usd"]
+
+        # Store prices in cache
+        cache.set("dgb_price_aud", str(dgb_price_aud), ttl_seconds = 14400)
```

Comment:
4 hours is a long time to cache the price - just wanted to make sure you meant 4 hours 😅

Reply:
No, that is a mistake! Well spotted! That should be 15 minutes! I'll push a fix!